Mjml.Net/Components/Body/SectionComponent.cs

```diff
@@ -1,11 +1,595 @@
+                .Attr("cellpadding", "0")
+                .Attr("cellspacing", "0")
+                .Attr("role", "presentation")
+                .Style("width", "100%")
+                .Style("border-radius", BorderRadius);
+
+            if (!isFullWidth)
+            {
+                if (hasBackground)
+                {
+                    tableElement
+                        .Style("background", GetBackground())
+                        .Style("background-color", BackgroundColor)
+                        .Style("background-position", BackgroundPosition)
+                        .Style("background-repeat", BackgroundRepeat)
+                        .Style("background-size", BackgroundSize);
+                }
+                else
+                {
+                    tableElement
+                        .Style("background", BackgroundColor)
+                        .Style("background-color", BackgroundColor);
+                }
+            }
+
+            renderer.ElementStart("tbody");
+            renderer.ElementStart("tr");
+            renderer.ElementStart("td")
+                .Style("border", Border)
+                .Style("border-bottom", BorderBottom)
+                .Style("border-left", BorderLeft)
+                .Style("border-right", BorderRight)
+                .Style("border-top", BorderTop)
+                .Style("direction", Direction)
+                .Style("font-size", "0px")
+                .Style("padding", Padding)
+                .Style("padding-bottom", PaddingBottom)
+                .Style("padding-left", PaddingLeft)
+                .Style("padding-right", PaddingRight)
+                .Style("padding-top", PaddingTop)
+                .Style("text-align", TextAlign);
+
+            renderer.StartConditionalTag();
+            renderer.ElementStart("table")
+                .Attr("border", "0")
+                .Attr("cellpadding", "0")
+                .Attr("cellspacing", "0")
+                .Attr("role", "presentation");
+            renderer.EndConditionalTag();
+
+            RenderChildren(renderer, ref context);
+
+            renderer.StartConditionalTag();
+            renderer.ElementEnd("table");
+            renderer.EndConditionalTag();
+
+            renderer.ElementEnd("td");
+            renderer.ElementEnd("tr");
+            renderer.ElementEnd("tbody");
+            renderer.ElementEnd("table");
+
+            if (hasBackground)
+            {
+                renderer.ElementEnd("div");
+            }
+
+            renderer.ElementEnd("div");
+        }
+
+        private static void RenderSectionEnd(IHtmlRenderer renderer)
+        {
+            renderer.StartConditionalTag();
+
+            renderer.ElementEnd("td");
+            renderer.ElementEnd("tr");
+            renderer.ElementEnd("table");
+
+            renderer.EndConditionalTag();
+        }
+
+        private void RenderSectionWithBackground(IHtmlRenderer renderer, ref GlobalContext context)
```

Comment:
I don't like this function implementation but it's very much a direct port of the original.

Reply:
It looks like a nightmare :D